File: cos/src/main/java/cos/olympus/game/events/Shot.java
package cos.olympus.game.events;

import cos.olympus.game.Creature;
import cos.ops.Direction;

public record Shot(
        @Override int id,
        int x,
        int y,
        int speed,
        Direction dir,
        int distance,
        int tick,
        @Override Creature source

) implements Spell, Event {
//
//    public static final boolean finished = ;


}

File: cos/src/main/java/cos/olympus/game/events/Shot.kt
package cos.olympus.game.events

import cos.olympus.game.Actor
import cos.olympus.game.Pos
import cos.olympus.game.Speed
import cos.olympus.game.TickId
import cos.ops.Direction

data class Shot(
    override val id: Int,
    val x: Pos,
    val y: Pos,
    val speed: Speed,
    val dir: Direction,
    val distance: Int,
    override val tick: TickId,
    override val source: Actor
) : Spell, Event
